src/credentials/examples/DeviceAttestationVerifierExample.cpp

```diff
@@ -0,0 +1,194 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "DeviceAttestationVerifierExample.h"
+
+#include <credentials/CHIPCert.h>
+#include <crypto/CHIPCryptoPAL.h>
+
+#include <lib/core/CHIPError.h>
+#include <lib/support/Span.h>
+#include <support/ScopedBuffer.h>
+
+using namespace chip::Crypto;
+
+namespace chip {
+namespace Credentials {
+namespace Examples {
+
+namespace {
+
+// Helper to do common logic to all providers
+CHIP_ERROR CopySpanToMutableSpan(ByteSpan span_to_copy, MutableByteSpan & out_buf)
+{
+    VerifyOrReturnError(IsSpanUsable(span_to_copy), CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(out_buf.size() >= span_to_copy.size(), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    memcpy(out_buf.data(), span_to_copy.data(), span_to_copy.size());
+    out_buf.reduce_size(span_to_copy.size());
+
+    return CHIP_NO_ERROR;
+}
```

Comment:
This is alrady in DeviceAttestationCredentialsProvider. This would benefit being moved to where ByteSpan is defined, as a utility used identically by any.

Reply:
agreed.